src/NzbDrone.Core/DecisionEngine/Specifications/ReleaseRestrictionsSpecification.cs

```diff
@@ -30,37 +31,83 @@ public virtual Decision IsSatisfiedBy(RemoteEpisode subject, SearchCriteriaBase
             _logger.Debug("Checking if release meets restrictions: {0}", subject);
 
             var title = subject.Release.Title;
-            var restrictions = _releaseProfileService.AllForTags(subject.Series.Tags);
+            List<ReleaseProfile> restrictions = _releaseProfileService.AllForTags(subject.Series.Tags);
 
-            var required = restrictions.Where(r => r.Required.IsNotNullOrWhiteSpace());
-            var ignored = restrictions.Where(r => r.Ignored.IsNotNullOrWhiteSpace());
-
-            foreach (var r in required)
+            foreach (ReleaseProfile restriction in restrictions)
             {
-                var requiredTerms = r.Required.Split(new[] {','}, StringSplitOptions.RemoveEmptyEntries).ToList();
+                // TODO: attach Enabled and IndexerId fields to restriction
+                /*if (!restriction.Enabled || restriction.IndexerId != subject.Release.IndexerId)
+                {
+                    continue;
+                }*/
+
+                var required = restrictions.Where(r => r.Required.IsNotNullOrWhiteSpace());
+                var ignored = restrictions.Where(r => r.Ignored.IsNotNullOrWhiteSpace());
+
+                var keyValueRegex = new Regex(@"\b\w+:\w+\b");
```

Comment:
- `private static readonly` please, so it can be reused.
- include RegexOptions.Compiled for the same purpose.
- Add the capture groups. (\w+):(\w+) (note that \w excludes _, so .* might be more appropriate)
 Avoids you having to do that Split(':') stuff later.
- Also just match it to the full string ^...$, no need for boundary anchors coz the terms are already split by `,`.